smart_open/s3.py

```diff
@@ -13,13 +13,17 @@
 import logging
 import warnings
 
+import boto
```

Comment:
Do we still need both `boto` and `boto3`? What's the relationship between the two (I forgot).

Reply:
We don't really need `boto`, IMHO. It's entirely separate to `boto3`.

Let's remove it in a separate PR.